codegenerator/cli/templates/static/codegen/src/ContractInterfaceManager.res

```diff
@@ -158,39 +146,6 @@ let getAllTopicsAndAddresses = (self: t): addressesAndTopics => {
   {addresses, topics}
 }
 
-let getLogSelection = (self: t): result<array<LogSelection.t>, exn> => {
-  try {
-    self.contractAddressMapping.addressesByName
-    ->Js.Dict.keys
-    ->Belt.Array.map(contractName => {
-      let interfaceOpt = self->getInterfaceByName(~contractName)
-      switch interfaceOpt {
-      | None => UndefinedInterface(contractName)->raise
-      | Some({interface}) => {
-          let topic0 = []
-          //Add the topic hash from each event on the interface
-          interface->Ethers.Interface.forEachEvent((eventFragment, _i) => {
-            topic0->Js.Array2.push(eventFragment.topicHash)->ignore
-          })
-
-          let topicSelection = LogSelection.makeTopicSelection(~topic0)->Utils.unwrapResultExn
-
-          let addresses = []
-          //Add the addresses for each contract
-          self.contractAddressMapping
-          ->ContractAddressingMap.getAddressesFromContractName(~contractName)
-          ->Belt.Array.forEach(address => addresses->Js.Array2.push(address)->ignore)
-
-          LogSelection.make(~addresses, ~topicSelections=[topicSelection])
-        }
-      }
-    })
-    ->Ok
-  } catch {
-  | exn => exn->Error
-  }
-}
```

Comment:
Sorry, it'll most likely conflict with your changes. But I hope the new version will be more convenient to work with (moved to HyperSync worker)